RHCLOUD-36959 | feature: improve logging for the Superkey worker

MikelAlejoBR opened the pull request:
Many of the current logs provided way too much information which made monitoring the service really hard. The goal of these changes is to provide that extra information only when an error occurs or when the "debug" log level is enabled.

For the rest of the informational logs, we are fine with just following the status of the worker.

## Jira ticket
[[RHCLOUD-36959]](https://issues.redhat.com/browse/RHCLOUD-36959)

MikelAlejoBR committed:
feature: improve logging for the Superkey worker

Many of the current logs provided way too much information which made
monitoring the service really hard. The goal of these changes is to
provide that extra information only when an error occurs or when the
"debug" log level is enabled.

For the rest of the informational logs, we are fine with just following
the status of the worker.

RHCLOUD-36959

dehort reviewed (commented):
Hey Mikel, looking good! 

A suggestion...instead of repeatedly adding the fields to the logger, how about creating and instance of the logger at the top of the method and adding fields to that instance as they become available.

Something like this:
```
log := l.Log.WithFields(logrus.Fields{"org_id": orgIdHeader})
// doing lots of cool stuff here...
// and here...
log.Errorf(`Logging the org_id here, even though its not on this line...look ma, no hands :)`, eventType)
```

This keeps you from repeatedly having to jam the same fields into the logger over and over and it decreases the risk of messing up the name of the field, etc.

I also like to build an instance of the logger and pass it along to other methods as they "work on the request" :smile: .  This helps you "stitch" together the logs in kibana and it keeps from having to repeatedly set the  fields...which is kinda gross (but logging is just kinda gross).  I have seen some code add the logger instance to a `context` and pass the context object around but that is gross too.  It ends up being a global and your code has to "dig" the logger out of the context object to use it.  It can be problematic if the code always expects the logger to be in the context, but the code was called with a context that does not have the logger.  Then the code blows up.  I usually see this in the tests / when writing tests.  You have to reverse engineer what all needs to be in the context to call the method.  Its kinda difficult and hard to follow.

MikelAlejoBR committed:
refactor: pass the context around for logging

Instead of repeating the fields so much, we can pass the context around
to reduce the code duplication and make that code easier to read and to
follow.

RHCLOUD-36959
